homeassistant/components/meater/sensor.py

```diff
@@ -34,20 +147,16 @@ def async_update_data():
 
         # Add entities for temperature probes which we've not yet seen
         for dev in devices:
-            if dev.id in known_probes:
+            if dev in known_probes:
                 continue
 
-            entities.append(
-                MeaterProbeTemperature(
-                    coordinator, dev.id, TemperatureMeasurement.Internal
-                )
-            )
-            entities.append(
-                MeaterProbeTemperature(
-                    coordinator, dev.id, TemperatureMeasurement.Ambient
-                )
+            entities.extend(
+                [
+                    MeaterProbeTemperature(coordinator, dev, sensor_description)
+                    for sensor_description in SENSOR_TYPES
+                ]
             )
-            known_probes.add(dev.id)
+            known_probes.add(dev)
```

Comment:
Why did we change this?

Reply:
Hmm, that does look like a bug, well spotted 👍

Reply:
Fix in https://github.com/home-assistant/core/pull/71283